Use simulate transaction jargon instead of preflight (#1381)
stellar/stellar-docs: docs/build/guides/transactions/simulateTransaction-Deep-Dive.mdx

Fees are calculated based on the resource consumption declared in the transaction. Refundable fees are charged before execution and refunded based on actual usage, while non-refundable fees are calculated from CPU instructions, read bytes, write bytes, and transaction size. [Check out this document for a more in depth understanding of fees and metering.](https://developers.stellar.org/docs/learn/fundamentals/fees-resource-limits-metering)

## Error Handling

> The simulate transaction mechanism provides an estimation of CPU and memory consumption in a transaction. It helps identify potential errors and resource limitations before actual submission. Errors returned by the host are propagated through RPC and do not cover network errors or errors in RPC itself.

Reviewer: “Transaction simulation mechanism” rolls better, not sure it’s a 1:1 replacement for “preflight” in that way.

> The transaction simulation mechanism provides an estimation of CPU and memory consumption in a transaction. It helps identify potential errors and resource limitations before actual submission. Errors returned by the host are propagated through RPC and do not cover network errors or errors in RPC itself.

Reviewer 2: ```suggestion
The transaction simulation mechanism provides an estimation of CPU and memory consumption of a contract invocation during a transaction. It also highlights [potential errors and resource limitations]((https://developers.stellar.org/docs/learn/encyclopedia/errors-and-debugging/debugging-errors#1-simulate-transaction)) of the invocation before actual submission. 
```

and could remove ln 382.
